src/dodal/devices/i10/id_apple2.py

```diff
@@ -0,0 +1,478 @@
+                f"Polarisation {pol} is not available:"
+                + f"/n Polarisations available:  {self._available_pol}"
+            )
+
+    @AsyncStatus.wrap
+    async def set(self, value: float) -> None:
+        """
+        Check polarisation state and use it together with the energy(value)
+        to calculate the required gap and phases before setting it.
+        """
+        if self.pol is None:
+            LOGGER.warning("Polarisation not set attempting to read from hardware")
+            pol, phase = await self.determinePhaseFromHardware()
+            if pol is None:
+                raise ValueError(f"Pol is not set for {self.name}")
+            else:
+                self.pol = pol
+
+        self._polarisation_set(self.pol)
+
+        gap, phase = self._get_id_gap_phase(value)
+        phase3 = -phase if self.pol == "la" else phase
+        id_set_val = Apple2Val(
+            top_outer=str(phase),
+            top_inner="0.0",
+            btm_inner=str(phase3),
+            btm_outer="0.0",
+            gap=str(gap),
+        )
+        LOGGER.info(f"Seting polarisation to {self.pol}, with {id_set_val}")
+        await self._set(value=id_set_val, energy=value)
+
+    async def _set(self, value: Apple2Val, energy: float) -> None:
+        """
+        Check ID is in a movable state and set all the demand value before moving.
+
+        """
+
+        # Only need to check gap as the phase motors share both fault and gate with gap.
+        await self.gap.check_id_status()
+        await asyncio.gather(
+            self.phase.top_outer.user_setpoint.set(value=value.top_outer),
+            self.phase.top_inner.user_setpoint.set(value=value.top_inner),
+            self.phase.btm_inner.user_setpoint.set(value=value.btm_inner),
+            self.phase.btm_outer.user_setpoint.set(value=value.btm_outer),
+            self.gap.user_setpoint.set(value=value.gap),
+        )
+        timeout = np.max(
+            await asyncio.gather(self.gap.get_timeout(), self.phase.get_timeout())
+        )
+        LOGGER.info(
+            f"Moving f{self.name} energy and polorisation to {energy}, {self.pol}"
+            + f"with motor position {value}, timeout = {timeout}"
+        )
+        await self.gap.set_move.set(value=1)
+        await wait_for_value(self.gap.gate, UndulatorGateStatus.close, timeout=timeout)
+        self._energy_set(energy)  # Update energy for after move for readback.
+
+    def _get_id_gap_phase(self, energy: float) -> tuple[float, float]:
+        """
+        Converts energy and polarisation to gap and phase.
+        """
+        gap_poly = self._get_poly(
+            lookup_table=self.lookup_tables["Gap"], new_energy=energy
+        )
+        phase_poly = self._get_poly(
+            lookup_table=self.lookup_tables["Phase"], new_energy=energy
+        )
+        return gap_poly(energy), phase_poly(energy)
+
+    def _get_poly(self, new_energy: float, lookup_table: dict) -> np.poly1d:
```

Comment:
should: can we get more on this lookup_table? it's a very special data structure and if we know just 'dict' then it's a bit nebulous what to expect later